query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PromQLQuery.java

```diff
@@ -0,0 +1,39 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Singular;
+
+@Getter
+@Builder(builderClassName = "Builder", buildMethodName = "build")
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class PromQLQuery {
+
+  @NonNull @Singular private List<String> queries;
+  private boolean isInstantRequest;
+  private Instant startTime;
+  private Instant endTime;
+  private Duration step;
```

Comment:
what is step ?

Reply:
A step is one of the arguments in PromQL range query API - https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries

So, when we create a time series request between time windows (start, end), step is param that helps create that series. So, you will have points at (start time, value), (star time + step, value), (start time + 2*step, value), so on

Reply:
We should try to be consistent within the code - especially within a single service. I think we call this period elsewhere - can we update one or the other so the two match?

Reply:
As discussed here (https://github.com/hypertrace/query-service/pull/111#discussion_r764060192), going ahead with `period` with comments.

Reply:
Done